src/nwb/NWBFile.cpp

```diff
@@ -28,22 +28,31 @@ constexpr SizeType SPIKE_CHUNK_XSIZE =
 
 std::vector<SizeType> NWBFile::emptyContainerIndexes = {};
 
-// NWBFile
+// Initialize the static registered_ member to trigger registration
+REGISTER_SUBCLASS_IMPL(NWBFile)
+
 NWBFile::NWBFile(std::shared_ptr<IO::BaseIO> io)
     : Container("/", io)
 {
 }
 
+NWBFile::NWBFile(const std::string& path, std::shared_ptr<IO::BaseIO> io)
+    : Container("/", io)  // Always use "/" for the path
+{
+  std::cerr << "NWBFile object is always the root. Path must be /" << std::endl;
+  assert(path == "/");
+}
+
 NWBFile::~NWBFile() {}
 
 Status NWBFile::initialize(const std::string& identifierText,
                            const std::string& description,
                            const std::string& dataCollection)
 {
   if (std::filesystem::exists(m_io->getFileName())) {
-    return m_io->open(false);
+    return m_io->open(IO::FileMode::ReadWrite);
   } else {
-    m_io->open(true);
+    m_io->open(IO::FileMode::Overwrite);
```

Comment:
```suggestion
    m_io->open();
```

Same as above. I don't think we need to specify the mode here, since the `open()` without parameters does this for us.

Reply:
thanks for catching! I missed that the file mode checking was duplicated here from `open()`